### PR TITLE
Optimize use of asBaseComponent in Picker and TextField

### PR DESCRIPTION
**Pull request by ethanshar**
## Description
Optimize use of asBaseComponent in Picker and TextField
Picker technically gets modifiers supports from TextField, so there's no need to use asBaseComponent there. 
I replaced it with `useThemeProps` hook to only add theme support

In TextField, I optimized the usage of asBaseComponent to get only relevant modifiers 
## Changelog
Optimize use of asBaseComponent in Picker and TextField

### CODE REVIEW

File: src/components/picker/index.tsx [hunk 1]
```diff
@@ -50,7 +49,9 @@ const DIALOG_PROPS = {
   height: 250
 };
 
-const Picker = (props: PropsWithChildren<PickerProps> & ForwardRefInjectedProps & BaseComponentInjectedProps) => {
+const Picker = (props: PropsWithChildren<PickerProps> & ForwardRefInjectedProps /*  & BaseComponentInjectedProps */
+) => {
```
Comment (Inbal-Tish) [inline src/components/picker/index.tsx:53]: No need to break line here... Why leave the commented BaseComponentInjectedProps? If we don't use it, let's delete it

**Review by Inbal-Tish — APPROVED** [review]

**Comment by Inbal-Tish** [comment]
@ethanshar Approved. Please see a small comment and check the failing build

File: src/components/picker/index.tsx (hunk 1 above)
Comment (ethanshar) [reply src/components/picker/index.tsx:53]: Fixed

**Comment by Inbal-Tish** [comment]
@ethanshar I enabled auto-merge, but worth taking a look there...

**Pull request merged**